src/main/java/core/basesyntax/JavaDateTimeApi.java

```diff
@@ -1,125 +1,177 @@
     /**
      * Given a random `someDate` date, return one of the following Strings:
-     *     - "`someDate` is after `currentDate`"
-     *                  if `someDate` is in the future relating to the `current date`;
-     *     - "`someDate` is before `currentDate`"
-     *                  if `someDate` is in the past relating to the `current date`;
-     *     - "`someDate` is today"
-     *                  if `someDate` is today;
+     * - "`someDate` is after `currentDate`"
+     * if `someDate` is in the future relating to the `current date`;
+     * - "`someDate` is before `currentDate`"
+     * if `someDate` is in the past relating to the `current date`;
+     * - "`someDate` is today"
+     * if `someDate` is today;
      */
     public String beforeOrAfter(LocalDate someDate) {
-        return someDate + "is today";
+        LocalDate now = LocalDate.now();
+        if (someDate.isAfter(now)) {
+            return someDate + " is after " + now;
+        } else if (someDate.isBefore(now)) {
+            return someDate + " is before " + now;
+        }
+        return someDate + " is today";
     }
 
     /**
      * Given a String representation of a date and some timezone,
      * return LocalDateTime in this timezone.
      */
     public LocalDateTime getDateInSpecificTimeZone(String dateInString, String zone) {
-        return LocalDateTime.now();
+        return ZonedDateTime.ofInstant(Instant.parse(dateInString), ZoneId.of(zone))
+                .toLocalDateTime();
     }
 
     /**
      * Given some LocalDateTime, return an OffsetDateTime with the local time offset applied
      * (`+02:00` for Ukraine).
-     *
+     * <p>
      * Example: we receive a LocalDateTime with a value `2019-09-06T13:17`.
-     *          We should return the OffsetDateTime with a value `2019-09-06T13:17+02:00`,
-     *          where `+02:00` is the offset for our local timezone.
-     *
+     * We should return the OffsetDateTime with a value `2019-09-06T13:17+02:00`,
+     * where `+02:00` is the offset for our local timezone.
+     * <p>
      * OffsetDateTime is recommended to use for storing date values in a database.
      */
     public OffsetDateTime offsetDateTime(LocalDateTime localTime) {
-        return OffsetDateTime.now();
+        return localTime.atOffset(ZoneId.of("Europe/Kiev").getRules().getOffset(Instant.now()));
     }
 
     /**
      * Given a String formatted as `yyyymmdd`,
      * return Optional of this date as a LocalDate.
      */
     public Optional<LocalDate> parseDate(String date) {
-        return Optional.empty();
+        int dateInt = Integer.parseInt(date);
+        int year = dateInt / 10000;
+        int month = (dateInt % 10000) / 100;
+        int day = dateInt % 100;
+        LocalDate localDate = null;
+        if (month > 0 && month <= MAX_MONTH && day > 0 && day <= MAX_DAY) {
+            localDate = LocalDate.of(year, month, day);
+        }
+        return Optional.ofNullable(localDate);
     }
```

Comment:
try to use here 'try-catch' it will make your code smaller

Reply:
now I know about BASIC_ISO_DATE)